src/compiler/parser.ts

```diff
@@ -277,6 +283,42 @@ function parseTextCommentNode(node: Node, ctx: ParsingContext): AST | null {
   return null;
 }
 
+function parseTCustom(node: Element, ctx: ParsingContext): AST | null {
+  if (!Object.keys(customDirectives).length) {
+    return null;
+  }
+  const nodeAttrsNames = node.getAttributeNames();
+  for (let attr of nodeAttrsNames) {
+    if (attr === "t-custom" || attr === "t-custom-") {
+      throw new OwlError("Missing custom directive name with t-custom directive");
+    }
+    if (attr.startsWith("t-custom-")) {
+      const directiveName = attr.split(".")[0].slice(9);
+      const customDirective = customDirectives[directiveName];
+      if (!customDirective) {
+        throw new OwlError(`Custom directive "${directiveName}" is not defined`);
+      }
+      const value = node.getAttribute(attr)!;
+      const modifier = attr.split(".").length > 1 ? attr.split(".")[1] : undefined;
+      node.removeAttribute(attr);
+      try {
+        node = customDirective(node, value, modifier);
+      } catch (error) {
+        throw new OwlError(
+          `Custom directive "${directiveName}" throw the following error: ${error}`
+        );
+      }
+      if (!(node instanceof Element)) {
```

Comment:
i am thinking that maybe, we want the allow a directive not to return anything (so it modifies in place the element)